Code/Legacy/CrySystem/XConsole.cpp

```diff
@@ -1682,11 +1682,12 @@ void CXConsole::DisplayHelp(const char* help, const char* name)
     else
     {
         char* start, * pos;
-        for (pos = strstr((char*)help, "\n"), start = (char*)help; pos; start = ++pos)
+        for (pos = strstr((char*)help, "\n"), start = (char*)help; pos; )
```

Comment:
This isn't the first time you have had to change this code.
The simplest solution is to switch this to use AZ::StringFunc::TokenizeVisitor and just delimite on new line
```
auto PrintLine = [](AZStd::string_view line)
{
    ConsoleLogInputResponse(" $3%.*s", AZ_STRING_ARG(line));
};
AZ::StringFunc::TokenizeVisitor(help, PrintLine, '\n');
```

Reply:
Right, I had posted it for review but messed up the PR and had to close it.  I opted here to just fix the bug by restoring original statement order, and think this is okay due to this being hardened legacy code.  If you feel strongly that I should change it to `TokenizeVisitor` I'll do it.

Reply:
"Hardened" is code for it's been around for a while, so we assume it works..
I suggest updating it to use TokenizeVisitor, so we never have to deal with this again.